lib/zone.ts

```diff
@@ -1229,40 +1223,11 @@ const Zone: ZoneType = (function(global: any) {
         global[symbolSetTimeout](drainMicroTaskQueue, 0);
       }
     }
-  }
-
-  function scheduleMicroTask(task: MicroTask) {
-    scheduleQueueDrain();
-    _microTaskQueue.push(task);
-  }
-
-  function consoleError(e: any) {
-    if ((Zone as any)[__symbol__('ignoreConsoleErrorUncaughtError')]) {
-      return;
-    }
-    const rejection = e && e.rejection;
-    if (rejection) {
-      console.error(
-          'Unhandled Promise rejection:',
-          rejection instanceof Error ? rejection.message : rejection, '; Zone:',
-          (<Zone>e.zone).name, '; Task:', e.task && (<Task>e.task).source, '; Value:', rejection,
-          rejection instanceof Error ? rejection.stack : undefined);
-    }
-    console.error(e);
-  }
-
-  function handleUnhandledRejection(e: any) {
-    consoleError(e);
-    try {
-      const handler = (Zone as any)[__symbol__('unhandledPromiseRejectionHandler')];
-      if (handler && typeof handler === 'function') {
-        handler.apply(this, [e]);
-      }
-    } catch (err) {
-    }
+    task && _microTaskQueue.push(task);
   }
 
   function drainMicroTaskQueue() {
+    const showError: boolean = !(Zone as any)[__symbol__('ignoreConsoleErrorUncaughtError')];
```

Comment:
Yes that sounds like a good idea. Perhaps we could do that as a second pass after this?